### DISCUSSION TITLE
custom directive

### DISCUSSION THREAD

**Discussion by bugsbugsbux** [executablebooks/meta#487] [Q&A] [answered]
hi, i'd like to create a new directive which transforms its content via an external program into svg. how do i hook into the directive parser?

i tried to adapt the containter plugin but it didnt give me access to the content, only to the first line ("info"). i also tried to grab the fences i want from the parsed nodes but even if i knew how to create a custom new token, i dont see how to render a list of tokens?
what confuses me when reading the sources of myst/docutils/markdown-it-py/mdit-py-plugins is that some functions take classes but most plugins dont explicitly create them but make a plugin function.
furthermore registering new plugins sometimes is done by adding *before* some other plugin to certain chains; how do i know which i want to put mine before and what if i want mine to be before several others (like fence and directive). as i understand it if i push to a chain my plugin will be last in the chain, not first...

am i just missing an easy way to subclass a directive which parses everything for me and i can just overwrite render and tell it to   replace itself with the output of an external program??

any help appreciated
kind regards

**Comment by michaelosthege** [comment, marked as answer]
Should be covered by this discussion: https://github.com/executablebooks/meta/discussions/655

**Reply by bugsbugsbux** [reply]
ok, thanks
